integreat_cms/api/v3/pages.py

```diff
@@ -20,25 +20,34 @@ def transform_page(page_translation):
     :param page_translation: single page translation object
     :type page_translation: ~integreat_cms.cms.models.pages.page_translation.PageTranslation
 
+    :raises ~django.http.Http404: HTTP status 404 if a parent is archived
+
     :return: data necessary for API
     :rtype: dict
     """
-    parent_page = page_translation.page.parent
-    if parent_page:
-        parent_absolute_url = parent_page.get_public_translation(
+    fallback_parent = {
+        "id": 0,
+        "url": None,
+        "path": None,
+    }
+
+    parent_page = page_translation.page.cached_parent
+    if parent_page and not parent_page.explicitly_archived:
+        parent_public_translation = parent_page.get_public_translation(
             page_translation.language.slug
-        ).get_absolute_url()
-        parent = {
-            "id": parent_page.id,
-            "url": settings.BASE_URL + parent_absolute_url,
-            "path": parent_absolute_url,
-        }
+        )
+        if parent_public_translation:
+            parent_absolute_url = parent_public_translation.get_absolute_url()
+            parent = {
+                "id": parent_page.id,
+                "url": settings.BASE_URL + parent_absolute_url,
+                "path": parent_absolute_url,
+            }
+        else:
+            raise Http404("No Page matches the given url or id.")
```

Comment:
```suggestion
            logger.info(
                "The parent %r of %r does not have a public translation in %r",
                parent_page,
                page_translation.page,
                page_translation.language,
            )
            raise Http404("No Page matches the given url or id.")
```
Since this error message is identical to the other ones that occur when the page really doesn't exist, I'd like to have a little more information to detect potential problems.